dash-pipeline/bmv2/stages/tunnel_stage.p4

```diff
@@ -0,0 +1,41 @@
+#ifndef _DASH_STAGE_TUNNEL_P4_
+#define _DASH_STAGE_TUNNEL_P4_
+
+control tunnel_stage(
+    inout headers_t hdr,
+    inout metadata_t meta)
+{
+    action set_tunnel_attrs(
+            @SaiVal[type="sai_ip_address_t"]
+            IPv4Address underlay_dip,
+            @SaiVal[type="sai_dash_encapsulation_t", default_value="SAI_DASH_ENCAPSULATION_VXLAN"]
+            dash_encapsulation_t dash_encapsulation,
+            bit<24> tunnel_key) {
+    push_action_static_encap(hdr = hdr,
+                            meta = meta,
+                            encap = dash_encapsulation,
+                            vni = tunnel_key,
+                            underlay_sip = hdr.u0_ipv4.src_addr,
+                            underlay_dip = underlay_dip,
+                            overlay_dmac = hdr.u0_ethernet.dst_addr);
```

Comment:
should we use the meta.overlay_data.dmac here, since it could be changed in the middle of the pipeline.

Reply:
At this point, it's the same value already